public/app/core/components/AccessControl/Permissions.tsx

```diff
@@ -91,13 +91,13 @@ export const Permissions = ({
 
   const onRemove = (item: ResourcePermission) => {
     let promise: Promise<void> | null = null;
-    if (item.userId) {
-      promise = setUserPermission(resource, resourceId, item.userId, EMPTY_PERMISSION);
-    } else if (item.teamId) {
-      promise = setTeamPermission(resource, resourceId, item.teamId, EMPTY_PERMISSION);
-    } else if (item.isServiceAccount && item.userId) {
-      promise = setUserPermission(resource, resourceId, item.userId, EMPTY_PERMISSION);
-    } else if (item.builtInRole) {
+    if (item.userUid && item.userUid !== '') {
```

Comment:
I think these empty string checks are redundant as e.g., `item.userUid` will cover that as well.

Reply:
Thanks, in the beginning it was not working because of the JSON to object parsing and I had these up trying to understand the problem.

Forgot to remove them